### ISSUE TITLE
Some problem about preparing dataset

### ISSUE THREAD

**Issue by tkuanlun350** [TuSimple/mx-maskrcnn#44]
Hello!  Thanks for your amazing work !

I want to prepare my dataset like Cityscapes, however I am confusing about the annotation in cityscapes dataset. In load_from_seg(rcnn/dataset/cityscape) function, what's the meaning of ins_id ? 

I tried to parse gtFine png file and found that each class is 1000*class_id and plus instance_id ? Is that correct ? ex: Label 23 have 8 instance, therefore the value in segmentation is [23000, 23001 ...23007], however, in _**aachen_000000_000019_gtFine_instanceIds.png**_ the ids of label 26 is 
[26003, 26004, 26005, 26006, 26007, 26008, 26009, 26010]

To sum up , I will appreciate if you can explain what is ins_id for and perhaps the annotation format of cityscapes (Cannot find detailed doc in official website :( ). Thank you very much.

**Comment by Zehaos** [comment]
Hi, @tkuanlun350 
 The meaning of pixel value of Instance seg image is exactly 1000*class_id + instance_id. 
_aachen_000000_000019_gtFine_instanceIds.png the ids of label 26 is [26003, 26004, 26005, 26006, 26007, 26008, 26009, 26010]_
is because of the ins_id [0,1,2] have been used.

**Comment by tkuanlun350** [comment]
Hi, Thanks for your explanation. However, I cannot understand why  ins_id [0,1,2]  were used ? the instance_id is not defined inside a single image ?
The instances in aachen_000000_000019_gtFine_instanceIds.png are listed below:
('ids', array([24000, 24001]))
('ids', array([25000]))
('ids', array([26003, 26004, 26005, 26006, 26007, 26008, 26009, 26010]))
('ids', array([33000, 33001]))

And I cannot found relevant code that use this field (ins_id) for training or testing. Can you explain where is field is used ?

Thanks for your kindly reply !

**Comment by Zehaos** [comment]
I must correct my answer, according to your supplement, ins_id [0,1,2] is missing in the label 26.
Here is code to extract instance in the ins seg image. Hope it help.
https://github.com/TuSimple/mx-maskrcnn/blob/master/rcnn/dataset/cityscape.py#L88-L106
https://github.com/TuSimple/mx-maskrcnn/blob/master/rcnn/processing/bbox_regression.py#L125-L145

**Comment by ChenFengYe** [comment]
I feel confused to this, so should I avoid ins_id[0,1,2] when generating my dataset?

**Comment by Zehaos** [comment]
@ChenFengYe You needn't.
